src/common/collection.c

```diff
@@ -1830,14 +1830,14 @@ static gchar *get_query_string(const dt_collection_properties_t property, const
       if(strcmp(operator, "[]") == 0)
       {
         if(number1 && number2)
-          query = g_strdup_printf("((%s >= %ld) AND (%s <= %ld))", colname, (long int)nb1, colname, (long int)nb2);
+          query = g_strdup_printf("((%s >= %ld) AND (%s <= %ld))", colname, (GTimeSpan)nb1, colname, (GTimeSpan)nb2);
```

Comment:
not sure it's needed to cast `nb1` and `nb2` as they are already defined as GTimeSpan
(same above)

Reply:
That seems logic ! :)
I do it.